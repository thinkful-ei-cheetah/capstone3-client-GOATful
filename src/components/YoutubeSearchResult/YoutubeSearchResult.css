.youtube-search-result {
  display: flex;
  /* min-width: 500px; */
  max-width: 65vw;
  padding: 8px;
  font-family: Roboto, Arial, sans-serif;
}

.youtube-search-result .left-col {
  flex: 1;
  position: relative;
}

.youtube-search-result > .left-col > img{
  width:246px;
  height: 138px;
  margin-right: 16px;
}

.youtube-search-result .left-col span {
  color: white;
  position: absolute;
  bottom: 10px;
  right: 20px;
  font-size: 12px;
  font-weight: 600;
  background-color: black;
  padding: 2px 3px;
}

.youtube-search-result .right-col {
  flex: 3;
  margin-left: 10px;
  overflow-x: hidden;
  overflow-y: hidden;
  max-height: 154px;
  width: 600px;
  letter-spacing: .5px;
}

.youtube-search-result .right-col  h2{
  margin: 0;
}

.right-col > h2 {
  font-size: 18px;
  overflow: hidden;
  font-weight: 500;
  text-overflow: ellipsis;
  line-height: 1.5rem;
  max-height: 48px;
}

.right-col > p {
  font-size: 13px;
  text-overflow: ellipsis;
}


.right-col > .sub-headings {
  color: #606060;
  margin: 0;
  padding: 0;
}

.right-col > .truncate{
  color: #606060;
  overflow: hidden;
  white-space: pre;
}

img.responsive-img {
  width: 100%;
  height: auto;
}

@media screen and (max-width: 425px) {
  .desktop-view-page > .youtube-search-result > .left-col > img {
    width: 200px;
    height: 118px;
  }

  .desktop-view-page > .youtube-search-result > .left-col > span {
    bottom: 40px;
  }
}

@media screen and (max-width: 570px) {
  .youtube-search-result .left-col span {
    bottom: 20px;
  }
}

.ellipsis-menu {
  visibility: hidden;
}