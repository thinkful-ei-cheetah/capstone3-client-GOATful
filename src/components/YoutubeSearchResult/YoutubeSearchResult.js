import React, {Component} from 'react';
import './YoutubeSearchResult.css';
import {
  randomIntFromInterval,
  shuffle,
  formatDuration,
  formatPublishedAtDate,
  formatViewCount
} from '../../Utils/Utils';
  
export default class YoutubeSearchResult extends Component {
  
  renderRandomDate = () => {
    const nums = ['2', '3', '4', '5', '6', '7', '8', '9'];
    const increments = ['minutes', 'months', 'years'];
    const num = shuffle(nums).pop();
    const increment = shuffle(increments).pop();
    return `${num} ${increment} ago`
  }

  renderRandomViews = () => {
    const num = randomIntFromInterval(1,999);
    return `${num}K`
  }

  render() {
    const {
      video_length, youtube_display_name, thumbnail_url, default_thumbnail, title, description, published_at, view_count
    } = this.props
    
    return (
      <div className='youtube-search-result'>
        <div className='left-col'>
          <img className='responsive-img' src={thumbnail_url || default_thumbnail} alt='youtube-thumbnail' />
          <span className='video-length'>{formatDuration(video_length)}</span>
        </div>

        <div className='right-col'>
          <h2>{title || 'this is a fake header'}</h2>
          <p className='sub-headings' >{youtube_display_name} • {formatViewCount(view_count)  || this.renderRandomViews()} • {formatPublishedAtDate(published_at) || this.renderRandomDate()}</p>
          <p className='truncate'>{description || 'this is a fake description'}</p>
        </div>
      </div>
    )
  }
}
  

