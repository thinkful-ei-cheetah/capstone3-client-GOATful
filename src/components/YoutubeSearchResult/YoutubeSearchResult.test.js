import React from 'react';
import ReactDOM from 'react-dom';
import YoutubeSearchResult from './YoutubeSearchResult';
import renderer from 'react-test-renderer';

describe('YoutubeSearchResult Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< YoutubeSearchResult />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const date = '2019-07-19T14:33:47.521Z'
    const tree = renderer
      .create(<YoutubeSearchResult published_at={date} view_count={100}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
