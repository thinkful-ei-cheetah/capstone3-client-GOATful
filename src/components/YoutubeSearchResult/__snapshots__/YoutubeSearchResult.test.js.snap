// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YoutubeSearchResult Component renders the UI as expected 1`] = `
<div
  className="youtube-search-result"
>
  <div
    className="left-col"
  >
    <img
      alt="youtube-thumbnail"
      className="responsive-img"
    />
    <span
      className="video-length"
    >
      0:00
    </span>
  </div>
  <div
    className="right-col"
  >
    <h2>
      this is a fake header
    </h2>
    <p
      className="sub-headings"
    >
      
       
      <span
        className="desktop-only"
      >
        •
      </span>
       
      <span
        className="mobile-clear"
      >
        100
         views • 
        11 hours ago
      </span>
    </p>
    <p
      className="truncate"
    >
      this is a fake description
    </p>
  </div>
  <div
    className="ellipsis-menu"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-ellipsis-v fa-w-6 "
      data-icon="ellipsis-v"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 192 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
